test/upgrade-e2e/post_upgrade_test.go

```diff
@@ -40,6 +40,17 @@ func TestClusterExtensionAfterOLMUpgrade(t *testing.T) {
 	t.Log("Wait for operator-controller deployment to be ready")
 	managerPod := waitForDeployment(t, ctx, "operator-controller-controller-manager")
 
+	t.Log("Wait for acquired leader election")
+	// Average case is under 1 minute but in the worst case: (previous leader crashed)
+	// we could have LeaseDuration (137s) + RetryPeriod (26s) +/- 163s
+	leaderCtx, leaderCancel := context.WithTimeout(ctx, 3*time.Minute)
```

Comment:
I am assuming 3 minutes is the worst case scenario. I am not familiar with `context.WithTimeout` , does it return if we acquire the lease before 163s?

Reply:
`context.WithTimeout` just gives you a context that timesout (gets cancelled) after then timeout period.
This means that the call to `watchPodLogsForSubstring(leaderCtx, managerPod, "manager", leaderSubstrings...)` will return with an error if it hasn't already after 3 minutes.

Reply:
ok, looks like it is a straight forward timeout method.